### PR TITLE
[TASK] Allow usernames with colon during login

### PR DESCRIPTION
**Pull request by AiNoKame**

### CODE REVIEW

File: app/scripts/modules/session/components/login-form.jsx [hunk 1]
```diff
@@ -64,7 +68,7 @@ var LoginForm = React.createClass({
     var _this = this;
 
     session.on('sync', function() {
-      _this.transitionTo('/payments/outgoing/all');
+      _this.transitionTo(_this.props.defaultRoute);
```
Comment (hserang) [inline app/scripts/modules/session/components/login-form.jsx]: if this is a default route, we should also set it as a default here using ```getDefaultProps()```, unless you want it to fail, which is alright if that's the intent.

File: app/scripts/shared/components/header/greeting.jsx [hunk 2]
```diff
@@ -3,26 +3,27 @@
...
+    userName: React.PropTypes.string
   },
 
   getDefaultProps: function() {
-    var defaults = {
-      greetingClassName: 'greeting-wrapper'
+    return {
+      greetingClassName: 'greeting-wrapper',
+      isLoggedIn: false,
+      userName: 'admin@example.com'
```
Comment (hserang) [inline app/scripts/shared/components/header/greeting.jsx]: I don't think we should use this as a default here. Perhaps userName should be just ```''``` if it's not set. We would likely be in a situation later wonder where/why admin@example.com is populating the field

File: app/scripts/shared/components/header/top-bar.jsx [hunk 3]
```diff
@@ -13,14 +13,18 @@ var TopBar = React.createClass({
       links: [{
         text: 'Login',
         href: '/login'
-      }]
+      }],
+      isLoggedIn: false,
+      userName: 'admin@example.com'
```
Comment (hserang) [inline app/scripts/shared/components/header/top-bar.jsx]: Same as above - I feel we should be deterministic about names and not setting them as defaults in multiple places

File: app/scripts/modules/session/models/session.js [hunk 4]
```diff
@@ -63,7 +63,8 @@ var Session = Backbone.Model.extend({
   },
 
   createCredentials: function(name, sessionKey) {
-    var encodedString = CryptoJS.enc.Base64.stringify(CryptoJS.enc.Utf8.parse(name + ':' + sessionKey));
+    var encodedString = CryptoJS.enc.Base64.stringify(
+      CryptoJS.enc.Utf8.parse(encodeURIComponent(name) + ':' + sessionKey));
```
Comment (hserang) [inline app/scripts/modules/session/models/session.js]: Great thinking, Rod!

**Comment by AiNoKame** [comment]
Thanks for the comments! I changed these up, accordingly

**Pull request merged**